### ISSUE TITLE
Data upload incorrectly succeeds on existing docid and revision

### ISSUE THREAD

**Issue by mbjones** [NCEAS/metacat#478]
---

Author Name: **Chris Jones** (Chris Jones)
Original Redmine Issue: 3197, https://projects.ecoinformatics.org/ecoinfo/issues/3197
Original Date: 2008-04-02
Original Assignee: Jing Tao

---

When uploading a data file to metacat in the case where the data document identifier already exists, the expected result is for metacat to 'update' the existing docid and revision, so long as the revision number of the new upload is greater than the existing revision number.

If the revision number is not incremented, but rather is identical to the existing revision number for the document, the upload incorrectly succeeds, but returns a upload size of zero bytes.  Here's an example:

upload of testdata.1.1:

<?xml version="1.0"?>
<success>
<docid>testdata.1.1</docid>
<size>515263</size>
</success>

re-upload a new revision of the data file as testdata.1.2:

<?xml version="1.0"?>
<success>
<docid>testdata.1.2</docid>
<size>515263</size>
</success>

Now, mistakenly upload the next revision without incrementing to testdata.1.3:

<?xml version="1.0"?>
<success>
<docid>testdata.1.2</docid>
<size>0</size>
</success>

The last upload should not succeed and should return an <error>.

**Comment by mbjones** [comment]
---

Original Redmine Comment
Author Name: **Jing Tao** (Jing Tao)
Original Date: 2008-04-03T16:12:11Z

---

When chris found this bug, he was testing metacat which had a 1.7.0 + plus version. Both chris and I tested current cvs head (and I tested 1.8.0 version) but we couldn't duplicate this bug. It seems the bug was fixed in 1.8.0 release. So I close this bug. Also, in junit test system, i added a test case for this scenario, so we can monitor this bug.
